tests/rptest/tests/cluster_config_test.py

```diff
@@ -1409,3 +1413,126 @@ def test_no_kafka(self):
+                 r"abs - .* Received .* AuthorizationFailure error response"
+             ])
+    @parametrize(cloud_storage_type=CloudStorageType.ABS)
+    def test_live_shared_key_change(self, cloud_storage_type):
+        """
+        This test ensures that 'cloud_storage_azure_shared_key' can
+        be safely updated without a full restart of the cluster.
+
+        The test performs the following steps:
+        1. Begin with a key in-place
+        2. Validate uploads work
+        3. Replace the key with a bogus one
+        4. Validate uploads are failing 
+        5. Set the key back to the initial value
+        6. Validate uploads work again
+        7. Try to unset the key
+        8. Validate that this is not allowed
+        """
+
+        initial_cloud_log_size = self.get_cloud_log_size()
+        self.produce_records(10, 1024)
+        wait_until(lambda: self.get_cloud_log_size() >= initial_cloud_log_size
+                   + 10 * 1024,
+                   timeout_sec=30,
+                   backoff_sec=5,
+                   err_msg="Data was not uploaded to cloud")
+
+        topic_leader_node = self.redpanda.partitions(self.topic)[0].leader
+        metric_check = MetricCheck(
+            self.logger,
+            self.redpanda,
+            topic_leader_node, [
+                "vectorized_cloud_storage_successful_uploads_total",
+                "vectorized_cloud_storage_failed_uploads_total"
+            ],
+            reduce=sum)
+
+        def check_uploads_failing():
+            return metric_check.evaluate([
+                ("vectorized_cloud_storage_successful_uploads_total",
+                 lambda a, b: b == a),
+                ("vectorized_cloud_storage_failed_uploads_total",
+                 lambda a, b: b > a)
+            ])
+
+        self.redpanda.set_cluster_config(
+            {
+                "cloud_storage_azure_shared_key":
+                "notakey02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="
+            },
+            expect_restart=False)
+
+        self.produce_records(10, 1024)
+        wait_until(check_uploads_failing,
+                   timeout_sec=30,
+                   backoff_sec=5,
+                   err_msg="Uploads did not fail")
+
+        self.redpanda.set_cluster_config(
+            {
+                "cloud_storage_azure_shared_key":
+                self.si_settings.cloud_storage_azure_shared_key
+            },
+            expect_restart=False)
+
+        initial_cloud_log_size = self.get_cloud_log_size()
+        self.produce_records(10, 1024)
+        wait_until(lambda: self.get_cloud_log_size() >= initial_cloud_log_size
+                   + 10 * 1024,
+                   timeout_sec=30,
+                   backoff_sec=5,
+                   err_msg="Data was not uploaded to cloud")
+
+        with expect_exception(requests.exceptions.HTTPError,
+                              lambda e: e.response.status_code == 400):
```

Comment:
this could probably be `with expect_http_error(400):`

Reply:
Wasn't aware of that api. Thanks for the suggestion.